alibi/explainers/cfrl_base.py

```diff
@@ -77,66 +81,69 @@ def __init__(self, size=1000):
             `size` * `batch_size`, where `batch_size` is inferred from the input tensors passed in the `append`
             method.
         """
-        self.x, self.x_cf = None, None           # buffers for the input and the counterfactuals
-        self.y_m, self.y_t = None, None          # buffers for the model's prediction and counterfactual target
-        self.z, self.z_cf_tilde = None, None     # buffers for the input embedding and noised counterfactual embedding
-        self.c = None                            # buffer for the conditional tensor
-        self.r_tilde = None                      # buffer for the noised counterfactual reward tensor
+        self.X: Optional[np.ndarray] = None              # buffer for the inputs
+        self.X_cf: Optional[np.ndarray] = None           # buffer for the counterfactuals
+        self.Y_m: Optional[np.ndarray] = None            # buffer for the model's prediction
+        self.Y_t: Optional[np.ndarray] = None            # buffer for the counterfactual targets
+        self.Z: Optional[np.ndarray] = None              # buffer for the input embedding
+        self.Z_cf_tilde: Optional[np.ndarray] = None     # buffer for the noised counterfactual embedding
+        self.C: Optional[np.ndarray] = None              # buffer for the conditional tensor
+        self.R_tilde: Optional[np.ndarray] = None        # buffer for the noised counterfactual reward tensor
```

Comment:
The `Optional` here are not really necessary as we have set "use strict optional" to `false` in mypy:
https://github.com/SeldonIO/alibi/blob/390a255403d61e8d7f87123f745b678b0a5e6753/setup.cfg#L24
https://mypy.readthedocs.io/en/stable/kinds_of_types.html#disabling-strict-optional-checking

Reply:
`Incompatible types in assignment (expression has type "ndarray", variable has type "None")`